CodeGeneration/Tests/ValidateSyntaxNodes/ValidateSyntaxNodes.swift

```diff
@@ -497,16 +506,190 @@ class ValidateSyntaxNodes: XCTestCase {
+        if case .node(kind: let kind) = child.kind, SYNTAX_NODE_MAP[kind]?.collectionNode != nil {
+          failures.append(
+            ValidationFailure(
+              node: node.kind,
+              message: "child '\(child.name)' is a SyntaxCollection but child is not marked as a collection"
+            )
+          )
+        }
+      }
+    }
+
+    assertFailuresMatchXFails(
+      failures,
+      expectedFailures: []
+    )
+  }
+
+  func testSyntaxCollectionChildrenArePlural() {
+    var failures: [ValidationFailure] = []
+
+    for node in SYNTAX_NODES.compactMap(\.layoutNode) {
+      for child in node.nonUnexpectedChildren where child.kind.isCollection {
+        if !child.name.hasSuffix("s") {
+          failures.append(
+            ValidationFailure(
+              node: node.kind,
+              message: "child '\(child.name)' is a collection and should thus be named as a plural"
+            )
+          )
+        }
+      }
+    }
+
+    assertFailuresMatchXFails(
+      failures,
+      expectedFailures: [
+        // The child is singular here, the path just consists of multiple components
+        ValidationFailure(node: .importDecl, message: "child 'Path' is a collection and should thus be named as a plural")
+      ]
+    )
+  }
+
+  /// Identifier is a wonderful ambiguous term. Almost always, 'name' or something similar is more expressive
+  func testNoChildContainsIdentifier() {
+    var failures: [ValidationFailure] = []
+
+    for node in SYNTAX_NODES.compactMap(\.layoutNode) {
+      for child in node.nonUnexpectedChildren {
+        if child.name.contains("Identifier") {
+          failures.append(
+            ValidationFailure(
+              node: node.kind,
+              message: "child '\(child.name)' should generally not contain 'Identifier'"
+            )
+          )
+        }
+      }
+    }
+
+    assertFailuresMatchXFails(
+      failures,
+      expectedFailures: [
+        // The identifier expr / pattern nodes do actually have a child that’s the identifier
+        ValidationFailure(node: .identifierExpr, message: "child 'Identifier' should generally not contain 'Identifier'"),
+        ValidationFailure(node: .identifierPattern, message: "child 'Identifier' should generally not contain 'Identifier'"),
+      ]
+    )
+  }
+
+  func testNoAbbreviationsInChildNames() {
+    var failures: [ValidationFailure] = []
+
+    let abbreviations = ["Args", "Params"]
+
+    for node in SYNTAX_NODES.compactMap(\.layoutNode) {
+      for child in node.nonUnexpectedChildren {
+        if abbreviations.contains(where: { child.name.contains($0) }) {
+          failures.append(
+            ValidationFailure(
+              node: node.kind,
+              message: "child '\(child.name)' should not contain abbreviations"
+            )
+          )
+        }
+      }
+    }
+
+    assertFailuresMatchXFails(
+      failures,
+      expectedFailures: []
+    )
+  }
+
+  func testChildrenDontEndWithExprEtc() {
```

Comment:
`testChildrenDontEndWithNodeKind`? Doesn't really matter though (and "Syntax" isn't a kind) 😅

Reply:
Yes, `Syntax` isn‘t a kind, children still shouldn’t end with `Syntax` 😉